File: srccode/frontend/src/services/feedbackService.js
import API from './api';

export const getAllFeedbacks = async () => {
  const response = await API.get('/feedbacks');
  return response.data;
};

export const getFeedbacksByCustomer = async (userId) => {
  const response = await API.get(`/feedbacks/customer/${userId}`);
  return response.data;
};

export const createFeedback = async (data) => {
  const response = await API.post('/feedbacks', data);
  return response.data;
};

export const updateFeedbackStatus = async (feedbackId, status) => {
  const response = await API.put(`/feedbacks/${feedbackId}/status`, {
    status
  });
  return response.data;
};

export const deleteFeedback = async (feedbackId) => {
  const response = await API.delete(`/feedbacks/${feedbackId}`);
  return response.data;
};

File: srccode/backend/restaurant_management_system/src/main/java/com/rms/restaurant_management_system/service/interfaces/FeedbackService.java
package com.rms.restaurant_management_system.service.interfaces;

import com.rms.restaurant_management_system.dto.request.FeedbackRequest;
import com.rms.restaurant_management_system.dto.request.UpdateFeedbackStatusRequest;
import com.rms.restaurant_management_system.dto.response.FeedbackResponse;

import java.util.List;

public interface FeedbackService {

    FeedbackResponse createFeedback(FeedbackRequest request);

    List<FeedbackResponse> getAllFeedbacks();

    List<FeedbackResponse> getFeedbacksByCustomer(Long userId);

    FeedbackResponse updateFeedbackStatus(Long feedbackId, UpdateFeedbackStatusRequest request);

    void deleteFeedback(Long feedbackId);
}